cmake/Platform/Mac/Install_mac.cmake

```diff
@@ -91,7 +94,7 @@ function(ly_install_target_override)
         ly_file_read(${LY_ROOT_FOLDER}/cmake/Platform/Mac/runtime_install_mac.cmake.in template_file)
         string(CONFIGURE "${template_file}" configured_template_file @ONLY)
         file(GENERATE
-            OUTPUT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${ly_platform_install_target_TARGET}.cmake
+            OUTPUT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${LY_BUILD_PERMUTATION}/${ly_platform_install_target_TARGET}.cmake
```

Comment:
The build permutation only needs to be taken into account when outputting a file into the CMAKE_INSTALL_PREFIX.

If a file is going into the CMAKE_BINARY_DIR, it doesn't need to worry about the build permutation as the CMAKE_BINARY_DIR is can only have one build_permutation.
i.e either a user generated a default Mac project using `cmake -B mac_xcode -S .` or the user generated a monolithic Mac project into a **different directory** using `cmake -B mac_code_monolithic -S.`

Reply:
Ah, ok. I will remove LY_BUILD_PERMUTATION then.

Reply:
Fixed